### PR TITLE
Initial commit

### LINKED ISSUE #112: src/std_misc+trait
1. [threads.md](https://github.com/nebula-labs/rust-by-example-vn/blob/main/src/std_misc/threads.md)
2. [clone.md](https://github.com/nebula-labs/rust-by-example-vn/blob/main/src/trait/clone.md)

### PR DESCRIPTION
**Pull request by NTP-996**
Closes: #112

### CODE REVIEW

**Comment by NTP-996** [comment]
fix #112

File: src/std_misc/threads.md [hunk 1]
```diff
@@ -1 +1,29 @@
 # Threads
+
+Rust cung cấp một cơ chế để sinh ra các luồng hệ điều hành gốc thông qua hàm `spawn`, đối số của hàm này là một bao đóng di động.
```
Comment (HongThaiPham) [inline src/std_misc/threads.md]:
```suggestion
Rust cung cấp một cơ chế để sinh ra các luồng của hệ điều hành gốc thông qua hàm `spawn`, đối số của hàm này là một bao đóng di động.
```

File: src/std_misc/threads.md [hunk 2]
```diff
@@ -1 +1,29 @@
...
+
+```rust,editable
+use std::thread;
+
+const NTHREADS: u32 = 10;
+
+// Đây là chủ đề `chính`
```
Comment (HongThaiPham) [inline src/std_misc/threads.md]:
```suggestion
// Đây là luồng `main`
```

File: src/std_misc/threads.md [hunk 3]
```diff
@@ -1 +1,29 @@
...
+fn main() {
+    // Tạo một vectơ children
```
Comment (HongThaiPham) [inline src/std_misc/threads.md]:
```suggestion
    // Tạo một vectơ để lưu trữ các luồng con được tạo ra
```

File: src/std_misc/threads.md [hunk 4]
```diff
@@ -1 +1,29 @@
...
+    }
+
+    for child in children {
+        // Chờ một luồng kết thúc để trả về kết quả.
+        let _ = child.join();
+    }
+}
+```
+
+Các chủ đề này sẽ được lên lịch bởi hệ điều hành.
```
Comment (HongThaiPham) [inline src/std_misc/threads.md]:
```suggestion
Các luồng này sẽ được lên lịch bởi hệ điều hành.
```

File: src/trait/clone.md [hunk 5]
```diff
@@ -1 +1,52 @@
 # Clone
+
+Khi xử lý tài nguyên, hành vi mặc định trong  chuyển chúng trong quá trình gán hoặc gọi hàm. Tuy nhiên, đôi khi chúng ta cũng cần phải tạo một bản sao của tài nguyên.
```
Comment (HongThaiPham) [inline src/trait/clone.md]:
```suggestion
Khi làm việc với các tài nguyên, hành vi mặc định là chuyển chúng trong quá trình gán hoặc gọi hàm. Tuy nhiên, đôi khi chúng ta cũng cần phải tạo một bản sao của tài nguyên.
```

File: src/trait/clone.md [hunk 6]
```diff
@@ -1 +1,52 @@
...
+
+[`Clone`][clone] giúp chúng ta thực hiện được điều này. Thông thường, chúng ta có thể sử dụng phương thức `.clone()` được xác định bởi đặc tính của `Clone`.
```
Comment (HongThaiPham) [inline src/trait/clone.md]:
```suggestion
Trait [`Clone`][clone] giúp chúng ta thực hiện được điều này. Thông thường, chúng ta có thể sử dụng phương thức `.clone()` được xác định bởi trait `Clone`.
```

File: src/trait/clone.md [hunk 7]
```diff
@@ -1 +1,52 @@
...
+
+```rust,editable
+// Một cấu trúc đơn vị không có tài nguyên
```
Comment (HongThaiPham) [inline src/trait/clone.md]:
```suggestion
// Một cấu trúc Unit không có tài nguyên
```

File: src/trait/clone.md [hunk 8]
```diff
@@ -1 +1,52 @@
...
+#[derive(Debug, Clone, Copy)]
+struct Unit;
+
+// Cấu trúc tuple với các tài nguyên triển khai đặc tính của `Clone`
```
Comment (HongThaiPham) [inline src/trait/clone.md]:
```suggestion
// Cấu trúc tuple với các tài nguyên triển khai trait `Clone`
```

File: src/trait/clone.md [hunk 9]
```diff
@@ -1 +1,52 @@
...
+#[derive(Clone, Debug)]
+struct Pair(Box<i32>, Box<i32>);
+
+fn main() {
+    // Khởi tạo `Đơn vị`
```
Comment (HongThaiPham) [inline src/trait/clone.md]:
```suggestion
    // Khởi tạo `Unit`
```

File: src/trait/clone.md [hunk 10]
```diff
@@ -1 +1,52 @@
...
+    let unit = Unit;
+    // Sao chép `Đơn vị`, không có tài nguyên nào để di chuyển
```
Comment (HongThaiPham) [inline src/trait/clone.md]:
```suggestion
    // Sao chép `Unit`, không có tài nguyên nào để di chuyển
```

File: src/trait/clone.md [hunk 11]
```diff
@@ -1 +1,52 @@
...
+    let copied_unit = unit;
+
+    // Cả hai `Đơn vị` đều có thể được sử dụng độc lập
```
Comment (HongThaiPham) [inline src/trait/clone.md]:
```suggestion
    // Cả hai `Unit` đều có thể được sử dụng độc lập
```

File: src/trait/clone.md [hunk 12]
```diff
@@ -1 +1,52 @@
...
+
+    // Khởi tạo `Pair`
+    let pair = Pair(Box::new(1), Box::new(2));
+    println!("original: {:?}", pair);
+
+    // Chuyển `pair` thành `moved_pair`, di chuyển tài nguyên
+    let moved_pair = pair;
+    println!("moved: {:?}", moved_pair);
+
+    // Lỗi! `pair` đã mất dữ liệu
```
Comment (HongThaiPham) [inline src/trait/clone.md]:
```suggestion
    // Lỗi! `pair` đã mất tài nguyên của nó do bị move
```

File: src/trait/clone.md [hunk 13]
```diff
@@ -1 +1,52 @@
...
+
+    // Cả hai `Đơn vị` đều có thể được sử dụng độc lập
+    println!("original: {:?}", unit);
+    println!("copy: {:?}", copied_unit);
+
+    // Khởi tạo `Pair`
+    let pair = Pair(Box::new(1), Box::new(2));
+    println!("original: {:?}", pair);
+
+    // Chuyển `pair` thành `moved_pair`, di chuyển tài nguyên
```
Comment (HongThaiPham) [inline src/trait/clone.md]:
```suggestion
    // Chuyển `pair` vào `moved_pair`, đồng thời cũng di chuyển tài nguyên của nó
```

File: src/trait/clone.md [hunk 14]
```diff
@@ -1 +1,52 @@
...
+    let moved_pair = pair;
+    println!("moved: {:?}", moved_pair);
+
+    // Lỗi! `pair` đã mất dữ liệu
+    //println!("original: {:?}", pair);
+    // TODO ^ Hãy thử chạy dòng lệnh trên
+
+    // Sao chép `moved_pair` vào `cloned_pair` (bao gồm tài nguyên)
+    let cloned_pair = moved_pair.clone();
+    // Xóa cặp ban đầu bằng cách sử dụng std::mem::drop
```
Comment (HongThaiPham) [inline src/trait/clone.md]:
```suggestion
    // Xóa pair gốc bằng cách sử dụng std::mem::drop
```

**Review by duonghb53 — APPROVED** [review]
LGTM

**Pull request merged**